Add Plugin: Rename File to Selection

AavaGames opened the pull request:
# I am submitting a new Community Plugin

## Repo URL
https://github.com/AavaGames/obsidian-rename-file-to-selection

<!--- Paste a link to your repo here for easy access -->
Link to my plugin:

## Release Checklist
- [x] I have tested the plugin on
  - [x]  Windows
  - [x]  macOS
  - [ ]  Linux
  - [x]  Android _(if applicable)_
  - [ ]  iOS _(if applicable)_
- [x] My GitHub release contains all required files (as individual files, not just in the source.zip / source.tar.gz)
  - [x] `main.js`
  - [x] `manifest.json`
  - [ ] `styles.css` _(optional)_
- [x] GitHub release name matches the exact version number specified in my manifest.json (_**Note:** Use the exact version number, don't include a prefix `v`_)
- [x] The `id` in my `manifest.json` matches the `id` in the `community-plugins.json` file.
- [x] My README.md describes the plugin's purpose and provides clear usage instructions.
- [x] I have read the developer policies at https://docs.obsidian.md/Developer+policies, and have assessed my plugins's adherence to these policies.
- [x] I have read the tips in https://docs.obsidian.md/Plugins/Releasing/Plugin+guidelines and have self-reviewed my plugin to avoid these common pitfalls.
- [x] I have added a license in the LICENSE file.
- [x] My project respects and is compatible with the original license of any code from other plugins that I'm using.
      I have given proper attribution to these other projects in my `README.md`.

Zachatoo commented:
[# obsidian-rename-file-to-selection](https://github.com/AavaGames/obsidian-rename-file-to-selection/blob/b46c549664f64b2baa56129c8204eefffc102576/README.md#L1)
Naming something "Obsidian xyz" is reserved for first party products we create, please change this header.

Please remove the styles.css file from your repo if you aren't using it.

[name: "Rename File To Selection",](https://github.com/AavaGames/obsidian-rename-file-to-selection/blob/b46c549664f64b2baa56129c8204eefffc102576/main.ts#L8)
[Use sentence case in UI](https://docs.obsidian.md/Plugins/Releasing/Plugin+guidelines#Use+sentence+case+in+UI)

Zachatoo commented:
Your `fundingUrl` links to an `itch.io` page, which in and of itself is fine, however the page doesn't explicitly mention how to donate. Please link to a section that explicitly mentions how to donate (updating your `itch.io` page, adding a section in your repo's README and linking to that instead, etc).

AavaGames commented:
Hi!

As per your request I have:
- removed styles.css
- removed obsidian from the readme title
- I changed the command name to sentence case from title case.
- add "If you would like to donate please PWYC for my game Datascape!" to my itch.io page

I hope that is sufficient. Let me know if there are any other changes you would like to see.

Thank you for your time!

joethei commented:
[Copyright (c) 2023 Aava Fertsman](https://github.com/AavaGames/obsidian-rename-file-to-selection/blob/5f70f22412180fdeeb89b4acd2481913bbe45ea4/License#L3)
Update this.

[editorCallback: (editor: Editor) => {](https://github.com/AavaGames/obsidian-rename-file-to-selection/blob/5f70f22412180fdeeb89b4acd2481913bbe45ea4/main.ts#L9)
This should be a `editorCheckCallback`, so that this code is no longer needed: [else if (sanitizedSelection != null || file != null)](https://github.com/AavaGames/obsidian-rename-file-to-selection/blob/5f70f22412180fdeeb89b4acd2481913bbe45ea4/main.ts#L22-L26)
